include/aikido/planner/vectorfield/BodyNodePoseVectorField.hpp

```diff
@@ -0,0 +1,53 @@
+#ifndef AIKIDO_PLANNER_VECTORFIELD_BODYNODEPOSEVECTORFIELD_HPP_
+#define AIKIDO_PLANNER_VECTORFIELD_BODYNODEPOSEVECTORFIELD_HPP_
+
+#include <dart/dynamics/BodyNode.hpp>
+#include <aikido/planner/vectorfield/MetaSkeletonStateSpaceVectorField.hpp>
+
+namespace aikido {
+namespace planner {
+namespace vectorfield {
+
+/// This class defines a vector field in a configuration space.
+///
+/// Any vector field should inherit this class to implememnt functions
+/// for calculating joint velocities giving joint positions and time,
+/// and checking planning status.
+class BodyNodePoseVectorField : public MetaSkeletonStateSpaceVectorField
+{
+public:
+  /// Constructor.
+  ///
+  /// \param[in] stateSpace MetaSkeleton state space.
+  /// \param[in] bn Body node of end-effector.
+  BodyNodePoseVectorField(
+      aikido::statespace::dart::MetaSkeletonStateSpacePtr stateSpace,
+      dart::dynamics::BodyNodePtr bodyNode);
+
+  // Documentation inherited.
+  virtual bool evaluateVelocity(
+      const aikido::statespace::StateSpace::State* state,
+      Eigen::VectorXd& qd) const override = 0;
```

Comment:
I was expecting this class to define `evaluateVelocity()` in terms of another function that returns the Cartesian velocity of the specified `BodyNode`. E.g.:
```c++
bool BodyNodePoseVectorField::evaluateVelocity(
      const aikido::statespace::StateSpace::State* state,
      Eigen::VectorXd& qd) const override = 0) override
{
  // Pseudocode. Actual implementation will require some massaging of data types.
  const auto cartesianState = mBodyNode->getPose(state);

  Eigen::Vector6d cartesianVelocity;
  if (!evaluateCartesianVelocity(cartesianState, cartesianVelocity))
    return false;

  return computeJointVelocityFromTwist(mBodyNode, cartesianVelocity, &qd);
}

// Implemented by the user:
virtual bool BodyNodePoseVectorField::evaluateCartesianVelocity(
  const Eigen::Isometry3d& _pose, Eigen::Vector6d& _cartesianVelocity) = 0;
```

In my opinion, the _whole point_ of this class is to allow the user to define a vector field over the pose of a `BodyNode` without worrying about the boilerplate required to call `computeJointVelocityFromTwist().